File: src/main/java/no/nav/pensjon/regler/domain/beregning2011/IJustering.java
package no.nav.pensjon.regler.domain.beregning2011;

import java.util.Date;

public interface IJustering {

    public abstract double getJusteringsfaktor();

    public abstract void setJusteringsfaktor(double justeringsfaktor);

    public abstract Date getJusteringTomDato();

    public abstract void setJusteringTomDato(Date justeringTomDato);
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/IJustering.kt
package no.nav.pensjon.regler.domain.beregning2011

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import java.util.*
import java.time.LocalDate

@JsonSubTypes(
    JsonSubTypes.Type(value = IFremskriving::class),
    JsonSubTypes.Type(value = ILonnsvekst::class)
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
interface IJustering {
    var justeringsfaktor: Double
    @Deprecated("Use justeringTomDatoLd instead")
    var justeringTomDato: Date?
    var justeringTomDatoLd: LocalDate?
}